domain/model/service/modelservice.go

```diff
@@ -94,5 +107,58 @@ func (s *ModelService) CreateModel(
 func (s *ModelService) DeleteModel(
 	ctx context.Context,
 ) error {
-	return s.st.Delete(ctx, s.modelID)
+	return s.modelSt.Delete(ctx, s.modelID)
+}
+
+// Status returns the status of the model.
```

Comment:
this should include the word "current" I think. For example "Status returns the current status of the model".

We should also document the possible error returns that could result from this function.

For example model not found is a likely scenario even if small.

Reply:
The state does not exist yet. We could probably assume the state should return a not found error at least.